zio-http/src/main/scala/zhttp/http/Request.scala

```diff
@@ -22,21 +30,67 @@ trait Request extends HeaderExtension[Request] { self =>
       override def url: URL                           = u
       override def getHeaders: Headers                = h
       override def remoteAddress: Option[InetAddress] = self.remoteAddress
-      override private[zhttp] def getBodyAsByteBuf    = self.getBodyAsByteBuf
+      override def unsafeBody(
+        msg: (
+          UnsafeChannel,
+          UnsafeContent,
+        ) => Unit,
+      ): Unit = self.unsafeBody(msg)
     }
   }
 
+  def unsafeBody(msg: (UnsafeChannel, UnsafeContent) => Unit): Unit
+
   /**
    * Decodes the content of request as a Chunk of Bytes
    */
-  def getBody: Task[Chunk[Byte]] =
-    getBodyAsByteBuf.flatMap(buf => Task(Chunk.fromArray(ByteBufUtil.getBytes(buf))))
+  def getBody: Queue[ByteBuf] = ???
+//    for {
+//      raw    <- ZStream.fromEffect(???)
+//      stream <- ZStream.fromQueue(raw)
+//    } yield stream
+
+  def getBodyAsStream: ZStream[Any, Throwable, ByteBuf] =
+    ZStream
+      .effectAsync[Any, Throwable, ByteBuf](cb =>
+        self.unsafeBody((ch, msg) => {
+          cb(IO.succeed(Chunk(msg.content.content())))
+          if (msg.isLast) {
+            ch.ctx.pipeline().remove(HTTP_CONTENT_HANDLER)
+            cb(IO.fail(None))
+          } else {
+            ch.read()
+          }
+
+        }),
+      )
```

Comment:
We can add a `getBodyChunk` which would look like this — 

```scala
def getBodyChunk: ZIO[Any, Option[Throwable], Chunk[Byte]]` = ???
```

This method will internally use `unsafeBody` and read the channel, one message at a time.